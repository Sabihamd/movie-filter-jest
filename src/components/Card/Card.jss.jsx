import { createUseStyles } from "react-jss";

const classes = createUseStyles({
  title: {
    fontSize: "15px",
    color: ({ theme }) => theme.color,
    fontFamily: "Lucida Sans",
     textTransform: "uppercase",
  },
  link: {
    extend: "title",
    textDecoration: "none",
  },
  cardContainer: {
    display: "flex",
    flexDirection: "column",
    padding: "30px",
    backgroundColor: "lightblue",
    "&:hover": {
      fontSize: "larger",
      fontFamily: "Times New Roman",
      fontWeight: "bolder",
      boxShadow: "5px 5px 5px 0px white",
    },
  },
  cardData: {
    display: "flex",
    flexDirection: "column",
    gap: "20px",
  },
});

export default classes;
