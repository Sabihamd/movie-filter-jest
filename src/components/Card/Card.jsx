import React from "react";
import { Link } from "react-router-dom";
import classes from "./Card.jss";
import { useTheme } from "react-jss";

export default function Card({ data }) {
  const theme = useTheme();
  const styles = classes({ theme });
  return (
    <div data-testid="card-container" className={styles.cardContainer}>
      <Link data-testid='link-test' className={styles.link} to="/movie" state={{ data }}>
        <div key={data.Title} className={styles.cardData}>
          <p className={styles.title}>{data.Title}</p>
          <img src={data.Image} alt="Movie" height="300px" width="290px" />
        </div>
      </Link>
    </div>
  );
}
