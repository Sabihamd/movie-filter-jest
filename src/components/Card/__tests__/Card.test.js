import { ThemeProvider } from "react-jss";
import {
  render,
  screen,
  cleanup,
  fireEvent,
  getByTestId,
} from "@testing-library/react";
import Card from "../Card";
import renderer from "react-test-renderer";
import { BrowserRouter } from "react-router-dom";

afterEach(() => {
  cleanup();
});

describe("testing Card Component", () => {
  const data = {
    Title: "The Lion King",
    Year: "2019",
    Runtime: "118 min",
    Image:
      "https://m.media-amazon.com/images/M/MV5BMjIwMjE1Nzc4NV5BMl5BanBnXkFtZTgwNDg4OTA1NzM@._V1_SX300.jpg",
    Poster:
      "https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcTVBpgqTaRffQeR19ukAV87es0Glp57oBLYKw&usqp=CAU",
    Plot: "Disney's The Lion King is about a young lion named Simba, who is the crown prince of an African Savanna. When his father dies in an accident staged by his uncle, Simba is made to feel responsible for his father's death and must overcome his fear of taking responsibility as the rightful heir to the throne.",
  };

  const theme = {
    backgroundColor: "#254D4A",
    color: "#254D4A",
  };

  test("should render Card Component and test the link", () => {
    render(
      <ThemeProvider theme={theme}>
        <BrowserRouter>
          <Card data={data} />
        </BrowserRouter>
      </ThemeProvider>
    );
    const card = screen.queryByTestId("card-container");
    expect(card).toBeInTheDocument();

    const link = screen.getByTestId("link-test");
    fireEvent.click(link);
    expect(window.location.pathname).toBe("/movie");
  });

  test("match the snapshot", () => {
    const card = renderer
      .create(
        <ThemeProvider theme={theme}>
          <BrowserRouter>
            <Card data={data} />
          </BrowserRouter>
        </ThemeProvider>
      )
      .toJSON();
    expect(card).toMatchSnapshot();
  });
});
