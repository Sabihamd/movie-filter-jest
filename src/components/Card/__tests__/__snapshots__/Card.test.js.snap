// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing Card Component match the snapshot 1`] = `
<div
  className="cardContainer-0-2-3"
  data-testid="card-container"
>
  <a
    className="link-0-2-2"
    data-testid="link-test"
    href="/movie"
    onClick={[Function]}
  >
    <div
      className="cardData-0-2-4"
    >
      <p
        className="title-0-2-1 title-d1-0-2-6"
      >
        The Lion King
      </p>
      <img
        alt="Movie"
        height="300px"
        src="https://m.media-amazon.com/images/M/MV5BMjIwMjE1Nzc4NV5BMl5BanBnXkFtZTgwNDg4OTA1NzM@._V1_SX300.jpg"
        width="290px"
      />
    </div>
  </a>
</div>
`;

exports[`testing Card Component should render Card Component 1`] = `
<div
  className="cardContainer-0-2-3"
  data-testid="card-container"
>
  <a
    className="link-0-2-2"
    href="/movie"
    onClick={[Function]}
  >
    <div
      className="cardData-0-2-4"
    >
      <p
        className="title-0-2-1 title-d0-0-2-5"
      >
        The Lion King
      </p>
      <img
        alt="Movie"
        height="300px"
        src="https://m.media-amazon.com/images/M/MV5BMjIwMjE1Nzc4NV5BMl5BanBnXkFtZTgwNDg4OTA1NzM@._V1_SX300.jpg"
        width="290px"
      />
    </div>
  </a>
</div>
`;

exports[`testing Card Component testing the Link 1`] = `
<div
  className="cardContainer-0-2-3"
  data-testid="card-container"
>
  <a
    className="link-0-2-2"
    href="/movie"
    onClick={[Function]}
  >
    <div
      className="cardData-0-2-4"
    >
      <p
        className="title-0-2-1 title-d1-0-2-6"
      >
        The Lion King
      </p>
      <img
        alt="Movie"
        height="300px"
        src="https://m.media-amazon.com/images/M/MV5BMjIwMjE1Nzc4NV5BMl5BanBnXkFtZTgwNDg4OTA1NzM@._V1_SX300.jpg"
        width="290px"
      />
    </div>
  </a>
</div>
`;
